src/msg/simple/Pipe.cc

```diff
@@ -828,16 +828,16 @@ void Pipe::set_socket_options()
     int flag = 1;
     int r = ::setsockopt(sd, IPPROTO_TCP, TCP_NODELAY, (char*)&flag, sizeof(flag));
     if (r < 0) {
-      r = -errno;
-      ldout(msgr->cct,0) << "couldn't set TCP_NODELAY: " << cpp_strerror(r) << dendl;
+      ldout(msgr->cct,0) << "couldn't set TCP_NODELAY: "
```

Comment:
this is intentional, as `ldout` and `<<` operator might change the errno, and they are evaluated before `cpp_strerror(errno)`, so we save the `errno` to `r` first.